doc/internationalization.md

```diff
@@ -0,0 +1,44 @@
+gRPC Internationalization
+=========================
+
+As a universal RPC framework, gRPC needs to be fully usable within/across different international environments. 
+This document describes gRPC API and behavior specifics when used in a non-english environment.
+
+## API Concepts
+
+While some API elements need to be able to represent non-english content, some are intentionally left as ASCII-only
+for simplicity & performance reasons.
+
+### Method name (in RPC Invocation)
+Method names are ASCII-only. Most gRPC services will use protobuf which only allows ASCII based method names anyway.
+Also, handling method names is a very hot code path.
+
+Recommended representation in language-specific APIs: string type.
+
+### Host name (in RPC Invocation)
+Host names are punycode encoded. Currently, the punycode needs to be provided by the user.
+
+Recommended representation in language-specific APIs: string/unicode string.
```

Comment:
To move on with this, I suggest just saying that the punycode needs to be user-provided. Punycode encoding by gRPC adds some extra complexity (at least in C core I assume) and no users have asked for this feature so far.
Later, if we see that the community demands support for IDN, we can introduce such support (e.g. either through adding a new API taking java.net.IDN as a parameter or as a utility/new function in C).
Any objections?